src/main/java/com/liferay/faces/util/cache/internal/CacheFactoryImpl.java

```diff
@@ -15,64 +15,60 @@
  */
 package com.liferay.faces.util.cache.internal;
 
+import javax.faces.context.ExternalContext;
+import javax.faces.context.FacesContext;
+
 import com.liferay.faces.util.cache.Cache;
 import com.liferay.faces.util.cache.CacheFactory;
+import com.liferay.faces.util.config.WebConfigParam;
 
 
 /**
  * @author  Kyle Stiemann
  */
 public class CacheFactoryImpl extends CacheFactory {
 
-	@Override
-	public <K, V> Cache<K, V> getCache() {
-		return new CacheImpl<K, V>();
-	}
+	// Private Final Data Members
+	private final int defaultInitialCacheCapacity;
 
-	@Override
-	public <K, V> Cache<K, V> getCache(int initialCapacity) {
+	public CacheFactoryImpl() {
 
-		validateInitialCapacity(initialCapacity);
+		FacesContext facesContext = FacesContext.getCurrentInstance();
```

Comment:
@stiemannkj1 If this is designed to be initialized during webapp initialization, then the variable should be named `startupFacesContext` in order to be consistent with the codebase. See the constructor of I18nImpl.java for an example.